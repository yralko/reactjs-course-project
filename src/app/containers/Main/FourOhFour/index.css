.pageNotFound {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translateX(-50%) translateY(-50%);
  font-size: 40px;
  text-align: center;
}
