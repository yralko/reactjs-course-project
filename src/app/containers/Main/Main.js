import React from 'react';
import { connect } from 'react-redux';
import SearchResults from '../../components/SearchResults';
import classes from './index.css';

const Main = (props) => {
  let renderedContent;

  if (props.fetchedFilms) {
    renderedContent = <SearchResults />;
  } else {
    renderedContent = <h1>Films not found</h1>;
  }

  return (
    <div className={classes.Main}>
      {renderedContent}
    </div>
  );
};

const mapStateToProps = (store) => {
  return {
    fetchedFilms: store.fetchedFilms,
  };
};

export default connect(mapStateToProps)(Main);
