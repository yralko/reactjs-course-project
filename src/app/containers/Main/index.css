.Main {
  flex: 1;
  position: relative;
}
