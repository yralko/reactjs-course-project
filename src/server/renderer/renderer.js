import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter as Router } from 'react-router-dom';
import App from '../../app/App';


export default (req, store, context) => {
  const root = (
    <Provider store={store}>
      <Router location={req.url} context={context}>
        <App />
      </Router>
    </Provider>
  );

  const content = renderToString(root); 

  return `
  <!doctype html>
  <html lang="en">
  <head>
    <meta charset="utf-8">
    <title>The HTML5 Herald</title>
    <base href="/" />
    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
    <div id="root">${content}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState())}</script>
    <script src="bundle.js"></script>
  </body>
  </html>
  `;
};
