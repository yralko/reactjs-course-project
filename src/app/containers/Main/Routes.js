import InitialPage from './InitialPage';
import ResultsPage from './ResultsPage';
import FourOhFour from './FourOhFour';
import SearchResults from '../../components/SearchResults';

export default [
  {
    path: '/',
    component: InitialPage,
    exact: true,
  },
  {
    path: '/movies',
    ...SearchResults,
  },
  {
    path: '*',
    component: FourOhFour,
  },
];
