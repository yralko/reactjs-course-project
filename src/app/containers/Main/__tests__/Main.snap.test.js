import React from 'react';
import renderer from 'react-test-renderer';
import { Main } from '../Main';

test('matches snapshot', () => {
  const tree = renderer.create(<Main />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
