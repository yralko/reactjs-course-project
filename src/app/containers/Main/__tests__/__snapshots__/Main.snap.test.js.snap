// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className={undefined}
>
  <h1>
    Films not found
  </h1>
</div>
`;
